### ISSUE TITLE
Handle room list updates properly with manual sort

### ISSUE THREAD

**Issue by kerryarchibald** [element-hq/element-web#25184]
Testing `ImportanceAlgorithm` showed that room updates are not handled properly for manual sorting and raise errors.
Skipped tests here:
https://github.com/matrix-org/matrix-react-sdk/blob/1440e347ea17a1f17ccefce3b9390fadacb3d5f7/test/stores/room-list/algorithms/list-ordering/ImportanceAlgorithm-test.ts#L95

**Comment by daniellekirkwood** [comment]
i dont think this is an issue in the new room list, closing but feel free to reopen with more details if i'm wrong!
